### DISCUSSION TITLE
LAB 05.01 -Task 1

### DISCUSSION THREAD

**Discussion by joanfco30** [rramosp/ai4eng.v1#124] [05 - GESTIÓN DE MODELOS]
Buenas tardes un cordial saludo para todos.  Mas que una inquietud quería hacer una consulta. Muchas uno de los restos en este curso es evitar usar los ciclos for ya que python a través de numpy permite recorrer array de forma mas optima. Pero muchas veces no encuentro la forma de hacerlo, como me ocurrió en este reto o primera tarea. 
![image](https://user-images.githubusercontent.com/60493083/167492016-bdecda96-4e4b-4534-8fe4-b450c3952834.png)
Esta tarea me funcionó, aun así me gustaría saber si pueden compartir la forma de hacerlo sin usar ciclos for. 
Cordial saludo

**Comment by Felipe-RA** [comment]
Saludos  joanfco30

Como bien dices, una de las ventajas de Python es su facilidad para manejar las operaciones vectoriales y matriciales.

Una de tantas posibles formas de implementar una solución a esta tarea con operaciones vectorizadas sería la siguiente:
```
trading_units = 1
converted_p = (p - 0.5) * 2
pnl =  ( (s[1:] - s[:-1]) * trading_units ) * converted_p   -   s[:-1] * trading_units * 0.01
```

Fijate en la última línea, la implementación es muy directa con respecto a la fórmula de pnl. 
La implementación sigue muy de cerca la definición matemática

Si tienes alguna pregunta con respecto al código, dimelo y lo miramos.

Hasta pronto!
